docs/arch/adr-010-new-versioning-scheme.md

```diff
@@ -0,0 +1,203 @@
+ │                │       │              │ │      │                │      │                │
+ │                │       │              │ │      │                │      │                │
+ │                │       │              │ │      │                │      │                │
+ │                │       │ v2022.11.01-0│ │      │                │      │                │
+ │                │       │              │ │      │                │      │                │
+ │                │       │              │ │      │                │      │                │
+ │                │       │              │ │      │                │      │                │
+ │                │       │ v2022.12.01-0│ │      │ v2022.12.01-0  │      │                │
+ │                │       │              │ │      │                │      │                │
+ │                │       │              │ │      │                │      │                │
+ │                │       │              ▼ │      │                │      │                │
+ │                │       │ v2023.01.01-0  ├──────► v2023.01.01-0  │      │                │
+ │                │       │                │      │              │ │      │                │
+ │                │       │                │      │              │ │      │                │
+ │                │       │                │      │              │ │      │                │
+ │                │       │                │      │ v2023.02.01-0│ │      │                │
+ │                │       │                │      │              │ │      │                │
+ │                │       │                │      │              │ │      │                │
+ │                │       │                │      │              │ │      │                │
+ │                │       │                │      │ v2023.03.01-0│ │      │                │
+ │                │       │                │      │              │ │      │                │
+ │                │       │                │      │              │ │      │                │
+ │                │       │                │      │              │ │      │                │
+ │                │       │                │      │ v2023.04.01-0│ │      │                │
+ │                │       │                │      │              │ │      │                │
+ │                │       │                │      │              │ │      │                │
+ │                │       │                │      │              ▼ │      │                │
+ │                │       │                │      │ v2023.05.01-0  ├──────► v2023.05.01-0  │
+ └────────────────┘       └────────────────┘      └────────────────┘      └────────────────┘
+```
+
+_Upgrade paths are supported on both horizontal and vertical axis, users can migrate from Kubernetes 1.24 to any other version but the upgrade will be executed sequentially._
```

Comment:
As discussed with @kyleprice here we may want to support diagonals as well: from the latest `installerVersion` on _Kubernetes Minor Version_ `1.24` to any other greater or equal `installerVersion` on _Kubernetes Minor Version_ `1.25`.

Reply:
This has already been addressed in the ADR.